### PR TITLE
Fixes #980 Expose C++ export for R

### LINKED ISSUE #980: Expose C++ export for R

### LINKED ISSUE #1113: Update to latest SimModel

### PR DESCRIPTION
**Pull request by Yuri05**
Fixes #1113 Update to latest SimModel

### CODE REVIEW

File: src/OSPSuite.R/Domain/SimulationBatch.cs [hunk 1]
```diff
@@ -69,12 +70,37 @@ ISimulationPersistableUpdater simulationPersistableUpdater
...
+      /// <param name="modelName">
+      ///   If empty (default): model will be named Standard and exported to Standard.cpp
+      ///   Otherwise: model will be named to "modelName" in the C++ code and exported to modelName.cpp.
+      ///              modelName must be both valid file name AND valid C++ identifier in such a case
+      /// </param>
+      public void ExportToCPPCode(string outputFolder, bool fullMode, string modelName="")
+      {
+         var exportMode = fullMode ? CodeExportMode.Formula : CodeExportMode.Values;
+         _simModelBatch.ExportToCPPCode(outputFolder, exportMode, modelName);
+      }
```
Comment (Yuri05) [inline src/OSPSuite.R/Domain/SimulationBatch.cs:102]: @PavelBal @msevestre **This** is the function which can be called from R to export the model (and variable parameters/initial values as they are set in the SimulationBatch) to C++.

With the current implementation of the SimulationBatch exporting in **Formula** and in **Value** mode would produce the same result I think (which corresponds to the Value mode) - which is ok anyway I think.

Exporting in **Formula** mode would make sense if e.g.
 - you would like to read the exported C++ code for whatever reason (I can hardly imagine any) or
 - if at the model creation time you don't know which parameters you will vary later and want to set ALL parameters as variable

Comment (PavelBal) [reply src/OSPSuite.R/Domain/SimulationBatch.cs:102]: @Yuri05 Could you please explain what 
` valid C++ identifier in such a case` is (sorry if it's a trivial question).

>if at the model creation time you don't know which parameters you will vary later and want to set ALL parameters as variable

Setting all parameters as variable has a negative impact on the performance?

Comment (Yuri05) [reply src/OSPSuite.R/Domain/SimulationBatch.cs:102]: >Could you please explain what

https://www.krellinst.org/archive/ais/textbook/unit7/CPP7.3.2.html

>Setting all parameters as variable has a negative impact on the performance?

Yes.

**Review by PavelBal — APPROVED** [review]

**Comment by Yuri05** [comment]
@msevestre Any concerns to merge this PR?

**Comment by msevestre** [comment]
Nope...i don't understand half of it but ...

**Pull request merged**